go/vt/vtgate/evalengine/fn_string.go

```diff
@@ -253,7 +259,7 @@ func (call *builtinASCII) compile(c *compiler) (ctype, error) {
 	switch {
 	case str.isTextual():
 	default:
-		c.asm.Convert_xc(1, sqltypes.VarChar, c.cfg.Collation, 0, false)
+		c.asm.Convert_xb(1, sqltypes.VarBinary, 0, false)
```

Comment:
This is `ASCII` but it didn't need to do actual character / collation aware conversion, just straight to binary is enough for `ASCII`.